modules/host/host.go

```diff
@@ -80,13 +73,32 @@ type Host struct {
 	secretKey  crypto.SecretKey
 
 	// File Management.
-	obligationsByID     map[types.FileContractID]*contractObligation
-	obligationsByHeight map[types.BlockHeight][]*contractObligation
+	obligationsByID map[types.FileContractID]*contractObligation
 
 	// Statistics
-	profit         types.Currency
-	fileCounter    int64
-	spaceRemaining int64
+	anticipatedRevenue types.Currency
+	fileCounter        int64
+	lostRevenue        types.Currency
+	revenue            types.Currency
+	spaceRemaining     int64
+
+	// RPC Tracking
+	atomicErroredCalls   uint64
+	atomicMalformedCalls uint64
+	atomicDownloadCalls  uint64
+	atomicRenewCalls     uint64
+	atomicReviseCalls    uint64
+	atomicSettingsCalls  uint64
+	atomicUploadCalls    uint64
+
+	// The resource lock is held by threaded functions for the duration of
+	// their operation. Functions should grab the resource lock as a read lock
+	// unless they are planning on manipulating the 'closed' variable.
```

Comment:
if you are holding `mu` alone it's not sufficient to guarantee that no new threads are going to be trying to grab resources - the resource lock is often grabbed before the thread is even initiated (though in hindsight, that might not be necessary). If the closing function is only holding `mu`, it might miss that another thread has grabbed the resource lock but not `mu`.

Basically, to guarantee safety you need to use the resource lock, and once you are using that there's no reason to also grab `mu`.